serpentTools/parsers/getmass.py

```diff
@@ -0,0 +1,141 @@
+# This python function can grab isotope masses from the NIST isotopic data file found in this directory.
+
+# definitely should have written something that uses regexes......too late :(
+import os
+# where is data file? This finds it.
+dir_path = os.path.dirname(os.path.realpath(__file__)) 
+datafile = dir_path + '/nistmasses.txt' # isotopic masses
+
+
+def getIsoMass(zaid):
+    """ grabs an isotopes mass in AMU. zaid should be in serpent format"""
+    zaid = str(zaid) #coerce as string. serves as error checker/conversion
+
+    #get z/a value
+    if len(zaid) ==4:
+        z=zaid[0]
+        a=zaid[-2:]
+        #remove leading zero if necessary
+        if a[0]=='0':
+            a=a[-1]
+    elif len(zaid) ==5:
+        z=zaid[:2]
+        a=zaid[-3:]
+        #leading zero maybe
+        if a[0]=='0':
+            a=a[-2:]
+    else:
+        raise Exception("Unidentifiable isotope {}".format(zaid) )
+
+    # check if the nuclide is natural
+    if '000' in zaid:
+        return getNaturalMass(zaid, z)
+
+    # now search the data file
+    # should be the isotope mass one from the NIST website
+    with open(datafile, 'r') as f:
+        current_z='  '
+        while current_z != z:
+            l=next(f)
+            current_z = l[-3:]
+            current_z = current_z[:2] #remove newline
+            current_z = current_z[-1] if current_z[0]==' ' else current_z
+            if z != '1':
+                while l != '\n' :
+                    try:
+                        l=next(f)
+                    except StopIteration:
+                        raise Exception("isotope {} not found".format(zaid))
+
+        # ok, now we're on the right Z val. now just search for the right A.
+        current_a = ''
+        while current_a != a:
+            try:
+                l=next(f)
+            except StopIteration:
+                return 0.0
+                #raise Exception('mass number not found for {}'.format(zaid))
+            if l[:4]=='Mass':
+                current_a = l[-4:]
+                current_a = current_a[:3] # remove \n
+                if current_a[:2]=='= ':
+                    current_a=current_a[-1]
+                elif current_a[0]==' ':
+                    current_a=current_a[1:]
+
+        # now finish the search by moving down one line.
+        l=next(f)
+        l=l.split()[4].split('(')[0]
+        mass=float(l)
+            
+
+
+    return mass 
+
+def getNaturalMass(zaid, z):
+    """gets mass of a natural isotope. this is really meant
+    to just be called by getIsoMass, so please use that function
+    instead. """
```

Comment:
What is the downside of calling this function directly? It seems like a shortcut if someone is looking for a specific type of isotope.

If this function is not supposed to be called, then:

- it should be [hidden](https://stackoverflow.com/a/4555970) with a single or double underscore.
- a message should be raised indicating this should not be used

Reply:
ah, thanks. The idea is the have the single function `getIsoMass` call `getNaturalMass` when natural ZAIDs get passed to `getIsoMass`. So, great point; I'll add the dunder.

Reply:
As for raising the message that "getNaturalMass" shouldn't be used, I think it would be best to just avoid this. The only good way I found to do this would be `import inspect` and check the stack frame. So, not adding this warning would mean less dependencies. It's up to you though.

Reply:
Using the dunder is fine. There should have been an or between those two options